Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/intersection.h

```diff
@@ -26,9 +26,1517 @@
+    }
+    //get a face-index map for mesh2
+    if(points_of_interest[mesh_id_2].size() == 0)
+    {
+      std::map<face_descriptor, int> fid_map;
+      int id = 0;
+      BOOST_FOREACH(face_descriptor fd, faces(mesh_2))
+      {
+        fid_map.insert(std::make_pair(fd,id++));
+      }
+      boost::associative_property_map< std::map<face_descriptor, int> >
+          fid_pmap(fid_map);
+      std::map<face_descriptor, int> fcc_map;
+
+      int nb_cc = Polygon_mesh_processing::connected_components(mesh_2,
+                                                                boost::make_assoc_property_map<std::map<face_descriptor, int> >(fcc_map),
+                                                                Polygon_mesh_processing::parameters::face_index_map(fid_pmap));
+      std::vector<bool> is_cc_treated(nb_cc, false);
+      int cc_treated = 0;
+      BOOST_FOREACH(face_descriptor fd, faces(mesh_2))
+      {
+        int cc=fcc_map[fd];
+        if(!is_cc_treated[cc])
+        {
+          points_of_interest[mesh_id_2].push_back(get(vpm2, target(halfedge(fd, mesh_2),mesh_2)));
+          is_cc_treated[cc] = true;
+          if(++cc_treated == nb_cc)
+            break;
+        }
+      }
+    }
+    //test if mesh_2 is included in mesh_1:
+
+    //for each CC, take a point on it and test bounded side
+    Side_of_triangle_mesh<TriangleMesh, GT, VPM> sotm(*trees[mesh_id_1], gt);
+    BOOST_FOREACH(CGAL::Point_3<GT> p, points_of_interest[mesh_id_2])
+    {
+      if(sotm(p) == CGAL::ON_BOUNDED_SIDE)
+      {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  template<class Mesh_box>
+  void operator()(const Mesh_box* b1, const Mesh_box* b2)
+  {
+    std::size_t mesh_id_1 = std::distance(meshes.begin(), b1->info());
+    std::size_t mesh_id_2 = std::distance(meshes.begin(), b2->info());
+
+
+    VPM vpm1 = choose_param(get_param(*(nps.begin() + mesh_id_1), internal_np::vertex_point),
+                            get_const_property_map(CGAL::vertex_point, *b1->info()));
+
+    VPM vpm2 = choose_param(get_param(*(nps.begin() + mesh_id_2), internal_np::vertex_point),
+                            get_const_property_map(CGAL::vertex_point, *b2->info()));
+
+    GT gt = choose_param(get_param(np, internal_np::geom_traits), GT());
+
+    //surfacic test
+    if(Polygon_mesh_processing::do_intersect(*b1->info(),
+                                             *b2->info(),
+                                             Polygon_mesh_processing::parameters::vertex_point_map(vpm1)
+                                             .geom_traits(gt),
+                                             Polygon_mesh_processing::parameters::vertex_point_map(vpm2)
+                                             .geom_traits(gt)))
+    {
+      *m_iterator++ = std::make_pair(mesh_id_1, mesh_id_2);
+    }
+    //volumic test
+    else if(test_volume)
+    {
+      if(is_mesh1_in_mesh2(*b1->info(), *b2->info(), mesh_id_1, mesh_id_2, vpm1, vpm2, gt))
```

Comment:
I think you can test first if the bboxes are disjoint

Reply:
like chjecking if any of the 8  points of b1 is inside b2, and if not, checking if any of the 8 points of b2 is in b1, and then decide if we perform the next tests, which will do that again ?

Reply:
I mean using [`do_overlap()`](https://doc.cgal.org/latest/Kernel_23/group__do__overlap__grp.html#ga6120c76991e6cae48af410ae07c0f2d7). If false the inclusion is obviously false.